src/constants.ts

```diff
@@ -42,16 +43,27 @@ export const DEFAULT_DELAY = 3000;
 export const DEFAULT_MAX_NUMBER_OF_TRIES = 10;
 export const DEFAULT_CREATE_CDA_TOKEN = true;
 
-export const VERSION_CONTENT_TYPE =
-  INPUT_VERSION_CONTENT_TYPE || DEFAULT_VERSION_CONTENT_TYPE;
-export const FEATURE_PATTERN = INPUT_FEATURE_PATTERN || DEFAULT_FEATURE_PATTERN;
-export const MASTER_PATTERN = INPUT_MASTER_PATTERN || DEFAULT_MASTER_PATTERN;
-export const VERSION_FIELD = INPUT_VERSION_FIELD || DEFAULT_VERSION_FIELD;
 export const DELETE_FEATURE = booleanOr(
   INPUT_DELETE_FEATURE,
   DEFAULT_DELETE_FEATURE
 );
+export const CREATE_CDA_TOKEN = booleanOr(INPUT_CREATE_CDA_TOKEN, DEFAULT_CREATE_CDA_TOKEN);
 export const SET_ALIAS = booleanOr(INPUT_SET_ALIAS, DEFAULT_SET_ALIAS);
+
+// set default actions to respect legacy input ENV's. This will make the new versikon backwards compatible.
+export const DEFAULT_ACTIONS = [
+  "createEnvironment",
+  CREATE_CDA_TOKEN && "createCDAToken",
+  "applyMigration",
+  SET_ALIAS && "updateAlias",
+  DELETE_FEATURE && "cleanUpEnvironments"
```

Comment:
not clear to me why we have some cases with a string constant and others with `<boolean> & <string constant>`.

shouldn't the default value be constant? and not be manipulated by user input? or are we using `DEFAULT_ACTIONS` as a way to store what the user wants to be executed?

---

looking at

```node
export const CREATE_CDA_TOKEN = booleanOr(INPUT_CREATE_CDA_TOKEN, DEFAULT_CREATE_CDA_TOKEN);
```

is seems we have 3 variables/constants:

- **DEFAULT_CREATE_CDA_TOKEN** for the default value
- **INPUT_CREATE_CDA_TOKEN** for the user input
- **CREATE_CDA_TOKEN** for the user input / default value evaluation

not saying we should use the same pattern. just pointing out that there is already a clear separation between constant for the default and variable / constant for what the GHA call should be doing.

Reply:
So, what's happened here is that I've deprecated SET_ALIAS, DELETE_FEATURE, CREATE_CDA_TOKEN. The DEFAULT_ACTIONS array is a replacement where you specify which actions you want on each run.

So in the new way to write it, you would list out the actions you want and skip set alias instead of using set_alias: false. This maintains backwards compatibility.